sentry-android-core/src/main/java/io/sentry/android/core/performance/AppStartMetrics.java

```diff
@@ -0,0 +1,186 @@
+
+  public static @NotNull AppStartMetrics getInstance() {
+
+    if (instance == null) {
+      synchronized (AppStartMetrics.class) {
+        if (instance == null) {
+          instance = new AppStartMetrics();
+        }
+      }
+    }
+    //noinspection DataFlowIssue
+    return instance;
+  }
+
+  public AppStartMetrics() {
+    appStartSpan = new TimeSpan();
+    sdkAppStartSpan = new TimeSpan();
+    applicationOnCreate = new TimeSpan();
+    contentProviderOnCreates = new HashMap<>();
+    activityLifecycles = new ArrayList<>();
+  }
+
+  /**
+   * @return the app start span Uses Process.getStartUptimeMillis() as start timestamp, which
+   *     requires API level 24+
+   */
+  public @NotNull TimeSpan getAppStartTimeSpan() {
+    return appStartSpan;
+  }
+
+  /**
+   * @return the app start time span, as measured pre-starfish Uses ContentProvider/Sdk init time as
+   *     start timestamp
+   */
+  public @NotNull TimeSpan getSdkAppStartTimeSpan() {
+    return sdkAppStartSpan;
+  }
+
+  public @NotNull TimeSpan getApplicationOnCreateTimeSpan() {
+    return applicationOnCreate;
+  }
+
+  public void setAppStartType(final @NotNull AppStartType appStartType) {
+    this.appStartType = appStartType;
+  }
+
+  public @NotNull AppStartType getAppStartType() {
+    return appStartType;
+  }
+
+  public boolean isAppLaunchedInForeground() {
+    return appLaunchedInForeground;
+  }
+
+  /**
+   * Provides all collected content provider onCreate time spans
+   *
+   * @return A sorted list of all onCreate calls
+   */
+  public @NotNull List<TimeSpan> getContentProviderOnCreateTimeSpans() {
+    final List<TimeSpan> measurements = new ArrayList<>(contentProviderOnCreates.values());
+    Collections.sort(measurements);
+    return measurements;
+  }
+
+  public @NotNull List<ActivityLifecycleTimeSpan> getActivityLifecycleTimeSpans() {
+    final List<ActivityLifecycleTimeSpan> measurements = new ArrayList<>(activityLifecycles);
+    Collections.sort(measurements);
+    return measurements;
+  }
+
+  public void addActivityLifecycleTimeSpans(final @NotNull ActivityLifecycleTimeSpan timeSpan) {
+    activityLifecycles.add(timeSpan);
+  }
+
+  public void clear() {
+    appStartType = AppStartType.UNKNOWN;
+    appStartSpan.reset();
+    sdkAppStartSpan.reset();
+    applicationOnCreate.reset();
+    contentProviderOnCreates.clear();
+    activityLifecycles.clear();
+  }
+
+  /**
+   * Called by instrumentation
+   *
+   * @param application The application object where onCreate was called on
+   * @noinspection unused
+   */
+  public static void onApplicationCreate(final @NotNull Application application) {
+    final long now = SystemClock.uptimeMillis();
+
+    final @NotNull AppStartMetrics instance = getInstance();
+    if (instance.applicationOnCreate.hasNotStarted()) {
+      instance.applicationOnCreate.setStartedAt(now);
+      instance.appLaunchedInForeground = ContextUtils.isForegroundImportance();
```

Comment:
`m`: I think this gives us some false positive or something, I recall report about app starts being unusually long, probably due to background starts. Should we take a chance and switch to Firebase's [approach](https://github.com/firebase/firebase-android-sdk/blob/28e3a6821910a0ae452fbcc5b173fafe919ddbaf/firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java#L552-L573)?